src/common/transformations/include/ov_ops/moe_expert.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <array>
+#include <memory>
+
+#include "openvino/core/node.hpp"
+#include "openvino/core/type/element_type.hpp"
+#include "openvino/op/op.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+namespace op {
+namespace internal {
+///
+/// \brief MOE experts
+class TRANSFORMATIONS_API MOEExpert : public ov::op::Op {
+public:
+    OPENVINO_OP("MOEExpert", "ie_internal_opset");
+
+    MOEExpert() = default;
+
+    struct Config {
+        size_t topk = 0;
+        size_t expert_num = 0;
+        size_t hidden_size = 0;
+        size_t intermediate_size = 0;
+        size_t fused_router_logic = false;
+        size_t group_size = 0;  // quantized group size, 0 for no group size. same for gate/up/down
+        ov::element::Type_t weight_type = ov::element::dynamic;  // same for gate/up/down
+        ov::element::Type_t scale_type = ov::element::dynamic;   // same for gate/up/down
+        ov::element::Type_t zp_type = ov::element::dynamic;      // same for gate/up/down
+        bool operator==(const Config& rhs) const {
+            return memcmp(this, &rhs, sizeof(*this)) == 0;
+        }
+    };
+
+    // 0: weight, 1: scale, 2: zp
+    struct ConstsPerExpert {
+        std::array<std::shared_ptr<ov::Node>, 3> gate;
+        std::array<std::shared_ptr<ov::Node>, 3> up;
+        std::array<std::shared_ptr<ov::Node>, 3> down;
+    };
+
+    MOEExpert(const OutputVector& args, const Config& config, const std::vector<ConstsPerExpert>& consts);
```

Comment:
This internal op, but usually the public ops has ctors with N Output<Node> as input.
The additional paramters are attributes for simple list is just paramter for more complex is helpers struct `Attributes` which hold them see Interpolate operator as example.

Then Op provide getter/setter to this attribute struct.

Reply:
Since it's an internal op and we know exactly how it will be used so no need to add not-use ctor with N output.
The `Config` and `ConstsPerExpert` are placed into `Attributes` for simpler interface.